components/raftstore/src/store/worker/compact.rs

```diff
@@ -301,16 +304,19 @@ where
         cf_name: &str,
         start_key: Option<&[u8]>,
         end_key: Option<&[u8]>,
+        bottommost_level_force: bool,
     ) -> Result<(), Error> {
         fail_point!("on_compact_range_cf");
         let timer = Instant::now();
         let compact_range_timer = COMPACT_RANGE_CF
             .with_label_values(&[cf_name])
             .start_coarse_timer();
-        box_try!(
-            self.engine
-                .compact_range_cf(cf_name, start_key, end_key, false, 1 /* threads */,)
-        );
+        box_try!(self.engine.compact_range_cf(
+            cf_name,
+            start_key,
+            end_key,
+            ManualCompactionOptions::new(false, 1, bottommost_level_force),
+        ));
         compact_range_timer.observe_duration();
         info!(
             "compact range finished";
```

Comment:
maybe we can add the filed in the log about `compact_option`.

Reply:
fine